pkg/homedir/homedir_linux.go

```diff
@@ -64,13 +64,14 @@ func stick(f string) error {
 
 // GetDataHome returns XDG_DATA_HOME.
 // GetDataHome returns $HOME/.local/share and nil error if XDG_DATA_HOME is not set.
+// If HOME and XDG_DATA_HOME are not set, passwd is consulted to determine the users home directory.
```

Comment:
```suggestion
// If HOME and XDG_DATA_HOME are not set, getpwent(3) is consulted to determine the users home directory.
```

It's more accurate to mention the system call, as NSS may be in play.

Reply:
Changed as requested in all three function